### ISSUE TITLE
Vercel function timeout (docs page generated at runtime)

### ISSUE THREAD

**Issue by MorelSerge** [fuma-nama/fumadocs#1068]
### To Reproduce

1. Setup a clean fumadocs repo
2. Install some content docs
3. Deploy to Vercel using a hobby plan (i.e. using lowest fn timeout and CPU/memory)

### Current vs. Expected behavior

The `/docs/[[slug]]/page.tsx` page seems to timeout on the default Vercel configuration, while the generated pages should be generated at build-time and therefore be served almost instantly from a CDN.

Instead, it seems to build at serve-time, timing out.
When increasing the timeout, it works (30s).

Additionally, due to this build at serve-time, in our setup, we reference an OpenAPI spec in a different package in our monorepo. When building at serve-time, this file is no longer available and it fails to serve the OpenApi pages.

### Provide environment information

```bash
Operating System:
  Platform: darwin
  Arch: arm64
  Version: Darwin Kernel Version 23.2.0: Wed Nov 15 21:53:18 PST 2023; root:xnu-10002.61.3~2/RELEASE_ARM64_T6000
  Available memory (MB): 32768
  Available CPU cores: 10
Binaries:
  Node: 22.1.0
  npm: 10.7.0
  Yarn: 1.22.21
  pnpm: 9.13.2
Relevant Packages:
  next: 15.0.3 // Latest available version is detected (15.0.3).
  eslint-config-next: 15.0.3
  react: 19.0.0-rc-66855b96-20241106
  react-dom: 19.0.0-rc-66855b96-20241106
  typescript: 5.5.3
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

Core, Integrations (e.g OpenAPI, Typescript DocsGen)

**Comment by fuma-nama** [comment]
You should see revalidate config, otherwise use an external url instead 

See https://fumadocs.vercel.app/docs/ui/openapi/configurations#input

**Comment by MorelSerge** [comment]
Hi @fuma-nama 
Appreciate the prompt reply, but disabling revalidate doesn't seem to stop Vercel from somehow trying to build it at runtime.

Any ideas? The same issue as before: timeout during first render + OpenAPI failing to render due to file not found at runtime

```ts
// Increasing max duration, since the default doesn't give Vercel enough time at run-time to compile.
export const maxDuration = 30;
// Vercel tries to rebuild this page at runtime, but that fails due to timeouts and OpenApi specs not found.
// There's no need to revalidate anyway, the files won't have changed since it's not a remote spec.
export const revalidate = false;
```

**Comment by fuma-nama** [comment]
Then you might need to use an external url. The timeout depends on the size of your schema, it's fast in general.

**Comment by MorelSerge** [comment]
So for anyone bumping on this issue, I figured it out by printing the `props` on the `/docs/[[...slug]]` page. It contained a dynamic rendering error because I used `cookies()`.

Didn't understand why I used `cookies()`, but completely forgot about my layout page for the docs used a gating mechanism to not allow visiting it when you did not have the right cookie set (some kind of authorization we built).
Removing that made the Fumadocs pages completely statically rendered at build-time again, removing all timeout and rendering issues mentioned earlier in this thread.
